### PR TITLE
Implement digital input

### PR DESCRIPTION
**Pull request by david-sawatzke**
Also adds an example, currently only for lpc845

### CODE REVIEW

**Comment by hannobraun** [comment]
Thank you! I didn't get to it today, but I intend to take a closer look at this tomorrow.

File: src/gpio.rs [hunk 1]
```diff
@@ -279,7 +281,105 @@ where
...
+    /// This method is only available, if the pin is in the GPIO state and the
+    /// pin is not already in input mode, i.e. the pin direction is output or
+    /// unknown. You can enter the GPIO state using [`Pin::into_gpio_pin`].
+    ///
+    /// Consumes the pin instance and returns a new instance that is in output
+    /// mode, making the methods to set the output level available.
+    ///
+    /// # Example
+    ///
+    /// # TODO
```
Comment (hannobraun) [inline src/gpio.rs]: Errant `TODO` spotted. Maybe referring to the example, which is out of date?

File: src/gpio.rs [hunk 2]
```diff
@@ -279,7 +281,105 @@ where
...
+    ///
+    /// // Transition pin into GPIO state, then set it to output
+    /// let mut pin = swm.pins.pio0_12
+    ///     .into_gpio_pin(&p.GPIO)
+    ///     .into_output();
+    ///
+    /// // Output level can now be controlled
+    /// pin.set_high();
+    /// pin.set_low();
+    /// ```
```
Comment (hannobraun) [inline src/gpio.rs]: This example is out of date. This should have been spotted by CI.

I'm fine with merging this as-is, and later fix the examples in one go. I've opened #192 to track this.

File: src/gpio.rs [hunk 3]
```diff
@@ -279,7 +281,105 @@ where
...
+    pub fn into_input(
+        self,
+    ) -> Pin<T, pin_state::Gpio<'gpio, direction::Input>> {
+        // TODO
```
Comment (hannobraun) [inline src/gpio.rs]: Errant `TODO`.

File: src/gpio.rs (hunk 1 above)
Comment (david-sawatzke) [reply src/gpio.rs]: The example was for the gpio output, but it's out of date as well

**Pull request merged**